src/presentation/http/middlewares/auth.ts

```diff
@@ -33,9 +36,33 @@ export default (): preHandlerHookHandler => {
     }
 
     /**
-     * TODO: verify token by specific service and pass payload to request
+     * Extract token from authorization header
      */
+    const token = authorizationHeader.replace('Bearer ', '');
 
-    done();
+    try {
+      const tokenPayload = await authService.verifyAccessToken(token);
+
+      /**
+       * Cast token payload to HttpRequest<Payload>
+       */
+      const params = request.params as HttpRequestParams<AuthPayload>;
+
+      /**
+       * Merge token payload with request params
```

Comment:
why?

Reply:
To pass this data to handler and use it. But I'm not sure, that it is good idea to pass data through the params. Maybe it is better to use `request.ctx` or something like this.

Reply:
yeah, adding a param looks strange

Reply:
use `routeConfig`